tmt/steps/provision/mrack.py

```diff
@@ -938,6 +945,34 @@ def create_host_requirement(self, host: CreateJobParameters) -> dict[str, Any]:
 
             # Whiteboard must be added *after* request preparation, to overwrite the default one.
             req['whiteboard'] = host.whiteboard
+            if host.bootc:
+                req['ks_append'] = req['ks_append'] if req.get('ks_append') else []
+                ks_meta = req['ks_meta'] if req.get('ks_meta') else ''
+                req['ks_meta'] = (
+                    'no_default_harness_repo disable_debug_repos disable_onerror ' + ks_meta
+                )
+                ks_list = [
+                    f"""
+ostreecontainer --url {host.bootc_image_url}
+%pre
+#!/bin/sh
+cat > /etc/ostree/auth.json <<EOF
+{host.auth_dict}
```

Comment:
And this would be `json.dumps(host.bootc_credentials)`. Convert on the boundary, when passing data from our side to 3d party API. We don't need this value to be a string, mrack does, we can and should keep the structure and only reduce it to an opaque string only when necessary.